pkg/gornir/runner.go

```diff
@@ -5,69 +5,39 @@ import (
 	"sync"
 )
 
-type JobParameters struct {
-	title  string
-	logger Logger
-	host   *Host
-}
-
-func NewJobParameters(title string, logger Logger) *JobParameters {
-	return &JobParameters{
-		title:  title,
-		logger: logger,
-	}
-}
-
-func (jp *JobParameters) Title() string {
-	return jp.title
-}
-
-func (jp *JobParameters) Logger() Logger {
-	return jp.logger
-}
-
-func (jp *JobParameters) Host() *Host {
-	return jp.host
-}
-
-func (jp *JobParameters) ForHost(host *Host) *JobParameters {
-	return &JobParameters{
-		title:  jp.title,
-		logger: jp.logger.WithField("host", host.Hostname),
-		host:   host,
-	}
-}
+// TaskInstanceResult is the resut of running a task on a given host
+type TaskInstanceResult interface{}
 
 // Task is the interface that task plugins need to implement.
 // the task is responsible to indicate its completion
 // by calling sync.WaitGroup.Done()
 type Task interface {
-	Run(context.Context, *sync.WaitGroup, *JobParameters, chan *JobResult)
+	Run(context.Context, Logger, *Host) (TaskInstanceResult, error)
 }
 
 // Runner is the interface of a struct that can implement a strategy
 // to run tasks over hosts
 type Runner interface {
-	Run(context.Context, Task, map[string]*Host, *JobParameters, chan *JobResult) error // Run executes the task over the hosts
-	Close() error                                                                       // Close closes and cleans all objects associated with the runner
-	Wait() error                                                                        // Wait blocks until all the hosts are done executing the task
+	Run(context.Context, Logger, Task, map[string]*Host, chan *JobResult) error // Run executes the task over the hosts
```

Comment:
Isn't `Logger` part of the Gornir?. You are also passing an `Inventory` (`map[string]*Host`), which is also in Gornir. I'm wondering if there is a cleaner way to pass all these values around. Probably something we can discuss in another PR or Issue.